### DISCUSSION TITLE
Urban Morphology: Morphometric Calculator (Point)

### DISCUSSION THREAD

**Discussion by Waqas-comsian** [UMEP-dev/UMEP#582] [General]
Please upload a tutorial for the Pre-processor>Urban Morphology> Morphometric Calculator (Point). It's really hard for a beginner to calculate the urban spatial factors such as FAI using UMEP.
I have DSM data and DEM data.

**Comment by biglimp** [comment]
You can find such instructions within this tuorial (https://umep-docs.readthedocs.io/projects/tutorial/en/latest/Tutorials/SuewsAdvanced.html#generating-data-from-the-geodatasets). 
Did you have a a look at the manual for this particular tool? There you can find information on what settings that are available (https://umep-docs.readthedocs.io/en/latest/pre-processor/Urban%20Morphology%20Morphometric%20Calculator%20%28Point%29.html)

**Comment by Waqas-comsian** [comment]
Thank you for your prompt reply. 
I am using the sample data of London (DSM, DEM, CDSM etc ) and the below settings, its still gives error. Can you help me solving the error.
Thank You again 

Algorithm 'Urban Morphology: Morphometric Calculator (Point)' starting…
Input parameters:
{ 'FILE_PREFIX' : 'aaa', 'INPUT_DEM' : 'D:/PhD Portfolio/PhD Portfolio/Objective 2-Urban Spatial Factors/Building Morphology/Data/Hong Kong/Waqas data/Extracted Stations/Extracted by polygone/FAI/DataSmallAreaLondon/DataSmallAreaLondon/DEM_LondonCity_1m.tif', 'INPUT_DISTANCE' : 200, 'INPUT_DSM' : 'D:/PhD Portfolio/PhD Portfolio/Objective 2-Urban Spatial Factors/Building Morphology/Data/Hong Kong/Waqas data/Extracted Stations/Extracted by polygone/FAI/DataSmallAreaLondon/DataSmallAreaLondon/DSM_LondonCity_1m.tif', 'INPUT_DSMBUILD' : 'D:/PhD Portfolio/PhD Portfolio/Objective 2-Urban Spatial Factors/Building Morphology/Data/Hong Kong/Waqas data/Extracted Stations/Extracted by polygone/FAI/DataSmallAreaLondon/DataSmallAreaLondon/DSM_LondonCity_1m.tif', 'INPUT_INTERVAL' : 5, 'INPUT_POINT' : '283962.437028,5712011.898867 [EPSG:32631]', 'INPUT_POINTLAYER' : None, 'OUTPUT_DIR' : 'TEMPORARY_OUTPUT', 'OUTPUT_POINT' : 'TEMPORARY_OUTPUT', 'OUTPUT_POLYGON' : 'TEMPORARY_OUTPUT', 'ROUGH' : 0, 'SAVE_POINT' : False, 'USE_DSM_BUILD' : True }

Point location obtained manually
x = 283962.437028
y = 5712011.898867
Traceback (most recent call last):
File "C:\Users/21120241r/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\preprocessor\imagemorphparmspoint_algorithm.py", line 204, in processAlgorithm
dsm = dataset.ReadAsArray().astype(float)
File "C:\PROGRA~1\QGIS33~1.1\apps\Python39\lib\site-packages\osgeo\gdal.py", line 3798, in ReadAsArray
from osgeo import gdal_array
File "C:\PROGRA~1/QGIS33~1.1/apps/qgis/./python\qgis\utils.py", line 888, in _import
mod = _builtin_import(name, globals, locals, fromlist, level)
File "C:\PROGRA~1\QGIS33~1.1\apps\Python39\lib\site-packages\osgeo\gdal_array.py", line 13, in <module>
from . import _gdal_array
File "C:\PROGRA~1/QGIS33~1.1/apps/qgis/./python\qgis\utils.py", line 888, in _import
mod = _builtin_import(name, globals, locals, fromlist, level)
ImportError: numpy.core.multiarray failed to import

Execution failed after 0.05 seconds

Loading resulting layers

**Reply by biglimp** [reply]
As you are using a DEM and DSM, you should not specify a dataset for **RasterDSM (only 3D objects)**. That option should only be used if you have a DSM with building height above ground. Leave **RasterDSM (only 3D objects)** empty and try again.

**Comment by Waqas-comsian** [comment]
![UMEP](https://github.com/UMEP-dev/UMEP/assets/73548724/e8ecb883-1411-4601-b9bc-661abe6c070c)
I am using above settings and it says 
No valid building DSM raster layer is selected
Execution failed after 0.02 seconds

**Reply by biglimp** [reply]
Sorry, I didnt read to the end of your first message. I have issues with version conflict between QGIS and numpy. You need to update either your QGIS or numpy (maybe both).
